pkg/e2e/infra/infra.go

```diff
@@ -328,75 +330,34 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
+		}, e2e.WaitLong, 5*time.Second).Should(BeTrue())
 
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[0].Name, scaleOut, 0))
+		By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[0].Name, scaleOut, 0))
 		err = scaleMachineSet(machineSets[0].Name, 0)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		Expect(err).NotTo(HaveOccurred())
 
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[1].Name, scaleOut, 0))
+		By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[1].Name, scaleOut, 0))
 		err = scaleMachineSet(machineSets[1].Name, 0)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("waiting for cluster to get back to original size. Final size should be %d nodes", existingClusterSize))
-		err = waitForClusterSizeToBeHealthy(client, existingClusterSize)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		Expect(err).NotTo(HaveOccurred())
 	})
 
-	g.It("drain node before removing machine resource", func() {
-		var err error
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("checking existing cluster size")
-		existingClusterSize, err := getClusterSize(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
+	It("drain node before removing machine resource", func() {
 		delObjects := make(map[string]runtime.Object)
 
-		defer func() {
-			// Remove resources
-			for _, obj := range delObjects {
-				if err := deleteObject(client, obj); err != nil {
-					glog.Errorf("[cleanup] error deleting object: %v", err)
-				}
-			}
-			// TODO(jchaloup): we need to make sure this gets called no matter what
-			// and waits until all labeled nodes are gone. Though, it it does not
-			// happend in the timeout set, it will not happen ever.
-			err := waitUntilNodesAreDeleted(client, []runtimeclient.ListOption{runtimeclient.MatchingLabels(nodeDrainLabels)})
-			o.Expect(err).NotTo(o.HaveOccurred())
-		}()
-
-		g.By("Taking the first worker machineset (assuming only worker machines are backed by machinesets)")
-		machinesets := mapiv1beta1.MachineSetList{}
-		err = wait.PollImmediate(e2e.RetryMedium, e2e.WaitShort, func() (bool, error) {
-			if err := client.List(context.TODO(), &machinesets); err != nil {
-				glog.Errorf("Error querying api for machineset object: %v, retrying...", err)
-				return false, nil
-			}
-			if len(machinesets.Items) < 1 {
-				glog.Errorf("Expected at least one machineset, have none")
-				return false, nil
-			}
-			return true, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("Creating two new machines, one for node about to be drained, other for moving workload from drained node")
+		By("Creating two new machines, one for node about to be drained, other for moving workload from drained node")
 		// Create two machines
 		randID := e2e.RandomString()
-		machine1 := machineFromMachineset(&machinesets.Items[0])
+		machine1 := machineFromMachineset(machineSet)
 		machine1.Name = "machine1-" + randID
 
```

Comment:
there's no need to create these machines now, you already got available the one created by the test scoped machineSet